src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -507,47 +561,114 @@ def regenerateDocumentation(self):
         #  in order for the documentation regeneration process to run.
         # The regen method is part of the documentation window. If the window is closed, the method no longer exists.
         if hasattr(self.parent, 'helpWindow'):
-            self.parent.helpWindow.regenerateHtml(self.filename)
+            self.parent.helpWindow.regenerateHtml(self.filename_py)
 
-    def canWriteModel(self, model=None, full_path=""):
+    def noModelCheckWarning(self):
+        """
+        Throw popup informing the user that no model checks will be run on a pure C model.
+        Ask user to acknowledge and give option to not display again.
+        """
+        msgBox = QtWidgets.QMessageBox(self)
+        msgBox.setIcon(QtWidgets.QMessageBox.Information)
+        msgBox.setText("No model checks will be run on your C file until a python file of the same name is created in your plugin directory.")
+        msgBox.setWindowTitle("No Python File Detected")
+        buttonContinue = msgBox.addButton("OK", QtWidgets.QMessageBox.AcceptRole)
+        doNotShowAgainCheckbox = QtWidgets.QCheckBox("Do not show again")
+        msgBox.setCheckBox(doNotShowAgainCheckbox)
+
+        msgBox.exec()
+
+        if doNotShowAgainCheckbox.isChecked():
+            # Update flag to not show popup again while this instance of TabbedModelEditor is open
+            self.showNoCompileWarning = False
+
+    def saveOverrideWarning(self, filename: str | Path, model_str: str):
+        """
+        Throw popup asking user if they want to save the model despite a bad model check.
+        Save model if user chooses to save, and do nothing if the user chooses to continue editing.
+
+        Returns True if user wanted to save file anyways, False if user wanted to continue editing without saving
+        """
+        msgBox = QtWidgets.QMessageBox(self)
+        msgBox.setIcon(QtWidgets.QMessageBox.Warning)
+        msgBox.setText("Model check failed. Do you want to save the file anyways?")
+        msgBox.setWindowTitle("Model Error")
+
+        # Add buttons
+        buttonContinue = msgBox.addButton("Continue editing", QtWidgets.QMessageBox.NoRole)
+        buttonSave = msgBox.addButton("Save anyways", QtWidgets.QMessageBox.AcceptRole)
+        # Set default button
+        msgBox.setDefaultButton(buttonContinue)
+
+        # Execute the message box and wait for the user's response
+        userChoice = msgBox.exec()
+
+        # Check which button was clicked and execute the corresponding code
+        if msgBox.clickedButton() == buttonContinue:
+            return False
+        elif msgBox.clickedButton() == buttonSave:
+            # Save files anyways
```

Comment:
a side effect of this check is that if I hit `Escape`, the whole dialog disappears without saving or without giving users a chance to do anything.

Reply:
I agree, but for the sake of the feature freeze, I've added this as something that needs fixing during the bug fix phase of the release.